momento-sdk/src/main/java/momento/sdk/responses/topic/TopicSubscribeResponse.java

```diff
@@ -0,0 +1,85 @@
+package momento.sdk.responses.topic;
+
+import java.util.function.Supplier;
+import momento.sdk.exceptions.SdkException;
+
+/** Represents the response for a topic subscribe operation. */
+public interface TopicSubscribeResponse {
+
+  /** Represents a successful topic subscribe operation. */
+  class Subscription implements TopicSubscribeResponse {
+    private SubscriptionState subscriptionState;
+
+    /**
+     * Constructs a Subscription instance with the provided subscription state.
+     *
+     * @param subscriptionState The subscription state.
+     */
+    public Subscription(SubscriptionState subscriptionState) {
+      super();
+      this.subscriptionState = subscriptionState;
+    }
+
+    /** Unsubscribes from the topic. */
+    public void unsubscribe() {
+      this.subscriptionState.unsubscribe();
+    }
+  }
+
+  /**
+   * Represents a failed topic subscribe operation. The response itself is an exception, so it can
+   * be directly thrown, or the cause of the error can be retrieved with {@link #getCause()}. The
+   * message is a copy of the message of the cause.
+   */
+  class Error extends SdkException implements TopicSubscribeResponse {
+
+    /**
+     * Constructs a topic subscribe error with a cause.
+     *
+     * @param cause The cause of the error.
+     */
+    public Error(SdkException cause) {
+      super(cause);
+    }
+  }
+
+  /**
+   * Returns the Subscription if the response is a Subscription, otherwise returns the provided
+   * default value.
+   *
+   * @param other The default value to be returned if the response is not a Subscription.
+   * @return The Subscription if the response is a Subscription, otherwise the default value.
+   */
+  default Subscription orElse(Subscription other) {
```

Comment:
In the thread I know we were suggesting implementing `orElse` but now that I'm looking at this I realize that it's extremely unlikely anyone would ever use this particular signature because they would need to have an existing `Subscription` object in scope to fall back to, and I can't really imagine that ever happening. I think we were advocating for the `Supplier` version of the signature so that the caller could pass in a lambda function that would do some kind of error handling and/or throw an exception. That's what `orElseGet` does, which you implemented below. So I think we should probably just remove this method for now.  @nand4011 ?